Update Key Vault documentation (#32313)
Azure/azure-sdk-for-net: sdk/keyvault/Azure.Security.KeyVault.Administration/README.md

To create a Managed HSM resource, run the following CLI command:

```PowerShell
az keyvault create --hsm-name <your-key-vault-name> --resource-group <your-resource-group-name> --administrators <your-user-object-id> --location <your-azure-location>
```

To get `<your-user-object-id>` you can run the following CLI command:

```PowerShell
az ad user show --id <your-user-principal> --query id
```

### Authenticate the client

In order to interact with the Azure Key Vault service, you'll need to create an instance of the client classes below. You need a **vault url**, which you may see as "DNS Name" in the portal,
and credentials to instantiate a client object.

The examples shown below use a [`DefaultAzureCredential`][DefaultAzureCredential], which is appropriate for most scenarios including local development and production environments.
Additionally, we recommend using a managed identity for authentication in production environments.
You can find more information on different ways of authenticating and their corresponding credential types in the [Azure Identity][azure_identity] documentation.

To use the `DefaultAzureCredential` provider shown below,

> or other credential providers provided with the Azure SDK, you must first install the Azure.Identity package:

Author: Markdown elides new lines in consecutive lines.

#### Create KeyVaultAccessControlClient

> Instantiate a `DefaultAzureCredential` to pass to the [KeyVaultAccessControlClient][rbac_client]. You can use the same instance of a token credential provider with multiple clients to cache credentials when appropriate.

Reviewer: ```suggestion
Instantiate a `DefaultAzureCredential` to pass to the [KeyVaultAccessControlClient][rbac_client]. The same instance of a token credential can be used with multiple clients if they will be authenticating with the same identity.
```
